radixdlt/src/integration/java/com/radixdlt/consensus/MultiNodeConsensusTest.java

```diff
@@ -0,0 +1,191 @@
+		VertexStore vertexStore
+	) {
+		Mempool mempool = mock(Mempool.class);
+		AtomicLong atomId = new AtomicLong();
+		doAnswer(inv -> {
+			Atom atom = mock(Atom.class);
+			when(atom.toString()).thenReturn(Long.toHexString(atomId.incrementAndGet()));
+			return Collections.singletonList(atom);
+		}).when(mempool).getAtoms(anyInt(), anySet());
+		ProposalGenerator proposalGenerator = new ProposalGenerator(vertexStore, mempool);
+		SafetyRules safetyRules = new SafetyRules(key, vertexStore, SafetyState.initialState());
+		PacemakerImpl pacemaker = new PacemakerImpl(quorumRequirements, Executors.newSingleThreadScheduledExecutor());
+		PendingVotes pendingVotes = new PendingVotes(quorumRequirements);
+		EventCoordinator eventCoordinator = new EventCoordinator(
+			proposalGenerator,
+			mempool,
+			testEventCoordinatorNetwork.getNetworkSender(key.getUID()),
+			safetyRules,
+			pacemaker,
+			vertexStore,
+			pendingVotes,
+			proposerElection,
+			key
+		);
+
+		return new ChainedBFT(
+			eventCoordinator,
+			testEventCoordinatorNetwork.getNetworkRx(key.getUID()),
+			pacemaker
+		);
+	}
+
+	private List<TestObserver<Vertex>> runBFT(
+		List<ECKeyPair> nodes,
+		QuorumRequirements quorumRequirements,
+		ProposerElection proposerElection
+	) {
+		final List<Pair<TestObserver<Vertex>, Observable<Event>>> bftEvents = nodes.stream()
+			.map(e -> {
+				RadixEngine radixEngine = mock(RadixEngine.class);
+				when(radixEngine.staticCheck(any())).thenReturn(Optional.empty());
+				VertexStore vertexStore = new VertexStore(genesisVertex, genesisQC, radixEngine);
+				TestObserver<Vertex> testObserver = TestObserver.create();
+				vertexStore.lastCommittedVertex().subscribe(testObserver);
+				ChainedBFT chainedBFT = createBFTInstance(e, proposerElection, quorumRequirements, vertexStore);
+				return new Pair<>(testObserver, chainedBFT.processEvents());
+			})
+			.collect(Collectors.toList());
+
+		Observable.merge(bftEvents.stream().map(Pair::getSecond).collect(Collectors.toList()))
+			.subscribe();
+
+		return bftEvents.stream()
+			.map(Pair::getFirst)
+			.collect(Collectors.toList());
+	}
+
+	@Test
+	public void given_3_correct_bft_instances_with_single_leader__then_all_instances_should_get_3_commits() throws Exception {
+		final List<ECKeyPair> nodes = Arrays.asList(new ECKeyPair(), new ECKeyPair(), new ECKeyPair());
+		final QuorumRequirements quorumRequirements = WhitelistQuorum.from(nodes.stream().map(ECKeyPair::getPublicKey));
+		final ProposerElection proposerElection = new Dictatorship(nodes.get(0).getUID());
+		final List<TestObserver<Vertex>> committedListeners = runBFT(nodes, quorumRequirements, proposerElection);
+
+		final int commitCount = 3;
+		for (TestObserver<Vertex> committedListener : committedListeners) {
+			committedListener.awaitCount(commitCount);
+			for (int i = 0; i < commitCount; i++) {
+				final int id = i;
+				committedListener.assertValueAt(i, v -> v.getAtom().toString().equals(Integer.toHexString(id)));
```

Comment:
This seems to rely on the behaviour of `Atom.toString`, so would be cleaner to get and compare the desired property (id) directly.

Reply:
Actually, scratch that since Atom is mocked anyway. 
Please add a comment for any future reader who might be confused by this